Bug: nil pointer dereference at migrator.go:336 in (Migrator).AlterColumn when schema field's DBName is absent from ColumnTypes() result

## GORM Playground Link

https://github.com/go-gorm/playground/pull/849

## Description

`(Migrator).AlterColumn` in `gorm.io/driver/postgres@v1.6.0` discards the error returned by `ColumnTypes()`, initializes `fieldColumnType` to `nil`, attempts to assign it inside a loop over the result set, and then dereferences it unconditionally — even when the loop completed without matching the schema field's `DBName`. Any caller that hands `AlterColumn` a field whose DB column does not exist (schema/DB drift, partially applied migration, race against a concurrent `DROP COLUMN`, etc.) crashes the process.

Source: https://github.com/go-gorm/postgres/blob/v1.6.0/migrator.go#L319-L336

```go
func (m Migrator) AlterColumn(value interface{}, field string) error {
    return m.RunWithValue(value, func(stmt *gorm.Statement) error {
        if field := stmt.Schema.LookUpField(field); field != nil {
            var (
                columnTypes, _  = m.DB.Migrator().ColumnTypes(value)  // err discarded
                fieldColumnType *migrator.ColumnType                  // nil
            )
            for _, columnType := range columnTypes {
                if columnType.Name() == field.DBName {
                    fieldColumnType, _ = columnType.(*migrator.ColumnType)
                }
            }

            fileType := clause.Expr{SQL: m.DataTypeOf(field)}
            isSameType := true
            // line 336 — NIL DEREF when no DB column matched the schema field
            if !strings.EqualFold(fieldColumnType.DatabaseTypeName(), fileType.SQL) {
```

### Trigger condition

1. `LookUpField(field)` returns non-nil → field exists in the model schema.
2. `ColumnTypes(value)` returns rows where NONE matches `field.DBName` — column is missing from the DB, OR `ColumnTypes` itself errored and the error was silently discarded.
3. Loop never assigns `fieldColumnType` → it stays `nil`.
4. Line 336 dereferences it → **panic**.

Originally suspected to be a `PrepareStmt + ALTER COLUMN TYPE` issue. Empirical testing rejected that hypothesis (5 strategies tried; none reproduced via the cached-plan invalidation path). The real cause is the missing-column case, which is far simpler.

### Reproduction (deterministic, single call)

Playground: https://github.com/go-gorm/playground/pull/849 — `TestBug_AlterColumnNilDerefOnMissingColumn` panics on every run.

```go
type Supplier struct {
    ID       uint   `gorm:"primaryKey"`
    Category string `gorm:"type:varchar(20);not null"`
}

// Create the table with ONLY "id" — no "category" column.
db.Exec(`DROP TABLE IF EXISTS suppliers`)
db.Exec(`CREATE TABLE suppliers (id bigserial PRIMARY KEY)`)

// Schema declares Category. LookUpField succeeds. ColumnTypes() returns
// only "id". Loop never assigns fieldColumnType. Line 336 panics.
_ = db.Migrator().AlterColumn(&Supplier{}, "Category")
```

Stack:

```text
panic: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
gorm.io/driver/postgres.Migrator.AlterColumn
        gorm.io/driver/postgres@v1.6.0/migrator.go:336
```

### Suggested fix

Two related defects in the same block — discarded `ColumnTypes` error, missing nil-guard. Both fixable in one patch while preserving the existing `resetPreparedStmts()` semantics:

```go
// gorm.io/driver/postgres/migrator.go:318-422
func (m Migrator) AlterColumn(value interface{}, field string) error {
    err := m.RunWithValue(value, func(stmt *gorm.Statement) error {
        if field := stmt.Schema.LookUpField(field); field != nil {
-           var (
-               columnTypes, _  = m.DB.Migrator().ColumnTypes(value)
-               fieldColumnType *migrator.ColumnType
-           )
+           columnTypes, err := m.DB.Migrator().ColumnTypes(value)
+           if err != nil {
+               return err
+           }
+           var fieldColumnType *migrator.ColumnType
            for _, columnType := range columnTypes {
                if columnType.Name() == field.DBName {
                    fieldColumnType, _ = columnType.(*migrator.ColumnType)
                }
            }
+           if fieldColumnType == nil {
+               return fmt.Errorf("postgres migrator: column %q not found in table %q for schema %T — schema/DB drift?",
+                   field.DBName, stmt.Table, value)
+           }

            fileType := clause.Expr{SQL: m.DataTypeOf(field)}
            // ... rest of function unchanged ...
        }
        return fmt.Errorf("failed to look up field with name: %s", field)
    })
    if err != nil {
        return err
    }
    m.resetPreparedStmts()  // existing — preserved
    return nil
}
```

### Prior reports of the same nil-`SQLColumnType` family

#5262 (closed 2022, "Panic on AutoMigration") describes the same `SQLColumnType nil` family from a slightly different call site (`Length()` instead of `DecimalSize()`). A community comment on that issue from `asherascout` diagnosed the exact mechanism:

> "when using select, some columns are missing in rows, so SQLColumnType may be nil"

That issue was closed without a code fix; only a "upgrade postgres to v1.3.4" was suggested, which did not actually patch the underlying nil deref — we still hit it on `v1.6.0`. The wrapper-side fix is the companion issue at https://github.com/go-gorm/gorm/issues/7764.

Related but distinct: #5737, #4588, #7255 cover the cached-plan invalidation family (different mechanism — pgx prepared-plan errors after DDL), not this nil-deref.

### Environment

- Go: 1.22.2
- gorm.io/gorm: v1.31.1
- gorm.io/driver/postgres: v1.6.0
- PostgreSQL: 16.4-alpine
- OS: Linux 6.18

--- a/migrator.go
+++ b/migrator.go
@@ -325,14 +325,18 @@ func (m Migrator) AlterColumn(value interface{}, field string) error {
 	err := m.RunWithValue(value, func(stmt *gorm.Statement) error {
 		if stmt.Schema != nil {
 			if field := stmt.Schema.LookUpField(field); field != nil {
-				var (
-					columnTypes, _  = m.DB.Migrator().ColumnTypes(value)
-					fieldColumnType *migrator.ColumnType
-				)
+				columnTypes, err := m.DB.Migrator().ColumnTypes(value)
+				if err != nil {
+					return err
+				}
+				var fieldColumnType *migrator.ColumnType
 				for _, columnType := range columnTypes {
 					if columnType.Name() == field.DBName {
 						fieldColumnType, _ = columnType.(*migrator.ColumnType)
 					}
+				}
+				if fieldColumnType == nil {
+					return fmt.Errorf("failed to find column type for field %s", field.DBName)
 				}
 
 				fileType := clause.Expr{SQL: m.DataTypeOf(field)}

